gcc/rust/lex/rust-lex.h

```diff
@@ -197,6 +198,9 @@ class Lexer
    * allocating new linemap */
   static const int max_column_hint = 80;
 
+  bool dump_lex_enabled;
+  std::ofstream dump_lex_out;
```

Comment:
This is good. I think the "best" way to go about this would be to keep a reference here, so 
```suggestion
  std::ofstream &dump_lex_out;
```

and to create the `ofstream` in the session manager.

This would allow all "dump operations" to stem from the same file and all "dump filenames" to reside in the same place

Reply:
What we could also do here is keep an `Optional<&std::ofstream>` and remove the `dump_lex_enabled` boolean. This way, we simply maybe keep a stream and dump to it when skipping tokens. This would also move the error handling (if the stream fails to open etc) to the session manager, where similar things already happen

Reply:
fixed